build-tools/src/main/java/org/elasticsearch/gradle/plugin/PluginBuildPlugin.java

```diff
@@ -51,6 +55,27 @@ public void apply(final Project project) {
             task.getOutputFile().set(file);
         });
 
-    }
+        project.getTasks().withType(GenerateTestBuildInfoTask.class).named("generateTestBuildInfo").configure(task -> {
+            var jarTask = project.getTasks().withType(Jar.class).named("jar").get();
+            String moduleName = (String) jarTask.getManifest().getAttributes().get("Automatic-Module-Name");
+            if (moduleName == null) {
+                moduleName = jarTask.getArchiveBaseName().getOrNull();
+            }
+            if (moduleName != null) {
+                task.getModuleName().set(moduleName);
```

Comment:
To avoid having to eagerly call `get()` above during configuration we could avoid doing so and instead use `Provider` across the board here like `task.getModuleName().set(jarTask.map(jar -> jar.getManifest().getAttributes().get("Automatic-Module-Name").toString()));`

Reply:
+1

Reply:
Looked at this with @mark-vieira and this isn't possible because we create a circular dependency since gradle assumes we want an output from the jar task implicitly. `getManifest()` also doesn't have a property so it can't be easily consumed in a lazy way. (Hope I'm not butchering what Mark told me here :) ).